### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by is0n** [codic12/worm#73]
1. Allow the user to change the mouse mod key used to resize/move windows (similar to [bspwm's](https://github.com/baskerville/bspwm/) `bspc config pointer_modifier`)
2. Focus windows when hovering over them (similar to [bspwm's](https://github.com/baskerville/bspwm/) `bspc config focus_follows_pointer`)
3. Cycle through windows using something like `Alt + Tab`

**Comment by codic12** [comment]
Thanks for the issue!

(1) pretty easy to implement ,definitely a good idea. same with (2) 
(3) might be kinda out of scope for this project; I use instead an external tool, https://github.com/sagb/alttab

**Comment by is0n** [comment]
> (1) pretty easy to implement ,definitely a good idea. same with (2)

Looking forward to it! 😀

> (3) might be kinda out of scope for this project; I use instead an external tool, https://github.com/sagb/alttab

This works perfectly! Thanks for the recommendation 👍🏽

**Comment by is0n** [comment]
Hey @codic12, any updates?

I've been tinkering around with implementing the `bspc config focus_follows_pointer` feature with shell scripting and I've been able to produce two scripts that sort of implement this feature.

Both scripts work but break whenever a new window is made or a different workspace is focused. This is however, caused by `wew` and not the actual script. Hopefully these scripts might help with implementing the feature :)

### Script 1:
```shell
# This script is mostly consistent but...
# it leaves the window border in the inactive state
wew -m 16 | while read; do
  id="$(xdotool getmouselocation | sed "s/.*://")"
  xdotool windowfocus "${id}"
done
```

### Script 2:
```shell
# This script is very inconsistent but when it works...
# it fully focus the window and activates the border
wew -m 16 | while read -r id; do
  pwid="$(echo "${id}" | awk '{print $2}')"
  cwid="$(xwininfo -children -id "${pwid}" | tail -n3 | head -n1 | awk '{ print $1 }')"
  wmctrl -i -a $cwid
done
```

**Comment by codic12** [comment]
Sorry for being so late! I do know how to implement focus follows mouse, I believe (XRaiseWindow && XSetInputFocus in EnterNotify, but I've been packing for going overseas. I'll try to get it done today.

**Comment by codic12** [comment]
Haven't really worked on it, but it should be straightfoward, just need to change one hardcoded thing to read from wormc.

**Comment by codic12** [comment]
> Sounds great! How is changing the default mouse modifier key going?

Done now! https://github.com/codic12/worm/commit/b4a1db098f98193bab70cb4daabb34de8224de89

**Comment by is0n** [comment]
It could just be my keyboard but it only works with the `Hyper`/`Mod3Mask`. I've tested it on both my work laptop and a fresh install of manjaro and have found no success.

`wormc` command I am using:
```shell
$ wormc modifier 32
```

If I want to use my super key, I have run the following commands...
```shell
$ xmodmap -e "remove mod4 = Super_L"
$ xmodmap -e "add mod3 = Super_L"
```
but in doing so, the super key becomes the hyper key and no longer behaves as the super key.

**Comment by codic12** [comment]
Did you try `wormc modifier 64` for changing it to Mod4Mask? That works fine for me.

**Comment by codic12** [comment]
Oh, i see my mistake, I accidentally typed 32 instead of 64 into the documentation :sweat_smile: fixed!

**Comment by is0n** [comment]
Yay! Using `wormc modifier 64` fixes the problem 🥳

**Comment by codic12** [comment]
Awesome, will try to implement focus-follows-mouse soon too.

**Comment by is0n** [comment]
After some more tinkering, I have created a script that fixes the problems from my previous scripts. It's not pretty but it works ¯\\\_(ツ)\_/¯

```shell
#!/usr/bin/env bash

xprop -spy -root  _NET_CLIENT_LIST | while read; do
  pgrep wew && killall wew
  wew -m 16 | while read WID; do
    PWID=$(echo $WID | awk '{ print $2 }')
    CWID=$(xwininfo -tree -id $PWID | awk '/":/ { print $1 }')
    xdo activate "$CWID"
  done &
done
```

Demo:

![Demo](https://user-images.githubusercontent.com/57725322/177024189-a4245344-8382-4b76-9f54-93cdc815dde2.gif)

Still excited for this to be a native feature! 😀

**Comment by codic12** [comment]
Awesome! I'm sorry, I haven't got around to implementing this since I've been stuck overseas with no access to my laptop for a while.I think focus follows mouse can be done by calling xraisewindow and xselectinput on EnterNotify events. Nice solution!

**Comment by codic12** [comment]
Finally solved by commit `5307384`!

**Comment by is0n** [comment]
I'm so glad that I can finally ditch my script! 🥳
